src/main/java/com/google/api/codegen/InterfaceView.java

```diff
@@ -28,10 +31,25 @@
   /** Gets the interfaces for the apis in the service config. */
   @Override
   public Iterable<Interface> getElementIterable(Model model) {
-    ArrayList<Interface> interfaces = new ArrayList<>();
+    List<Interface> interfaces = new ArrayList<>();
     for (Api api : model.getServiceConfig().getApisList()) {
       interfaces.add(model.getSymbolTable().lookupInterface(api.getName()));
     }
     return interfaces;
   }
+
+  /**
+   * Helper to extract the types from the underlying model.
+   *
+   * @see com.google.api.Service#getTypesList()
+   * @param model model with service config
+   * @return types
+   */
+  public Iterable<TypeRef> getTypes(Model model) {
```

Comment:
- Suggest renaming this to `getAdditionalTypes`. (This only returns the types listed in the `types` section of the service config, not all types in the Model's SymbolTable, right?)
- I think this code probably belongs in `ProtoApiModel`, rather than here. These `*View` classes aren't really designed to implement utility methods outside of `getElementIterable`.

Reply:
Sure, seems reasonable.